### DISCUSSION TITLE
RuneMaster: Port to raylib-apl

### DISCUSSION THREAD

**Discussion by MaxCan-Code** [Dyalog/GSOC#4] [Ideas]
(format based on https://swift.org/gsoc2024)
**Project size**: ?? hours
**Estimated difficulty**: 4/5
**Recommended skills**: APL, game programming

**Description**
[RuneMaster](https://github.com/abrudz/runemaster) is maze game for learning APL on the web. [raylib-apl](https://github.com/Brian-ED/raylib-apl) is a library made to write cross-platform graphical applications using the Dyalog APL programming language. This project will be implementing RuneMaster in raylib-apl.

**Expected outcomes**:
- A RuneMaster made in raylib-apl
- Allow user-defined levels and questions
- Having a live APL session running in game

**Mentors**:
??

**Comment by sloorush** [comment]
Thank you, Max!

cc: @abrudz @rikedyp

**Comment by abrudz** [comment]
Right now:
* RuneMaster has an (almost finished) implementation in modern front-end vanilla JS with no dependencies besides using the online TryAPL API, and therefore is inherently cross-platform while requiring no installation.
* raylib-apl [only supports Linux](https://github.com/Brian-ED/raylib-apl?tab=readme-ov-file#documentation) and is strictly a desktop interface that requires an installed Dyalog APL at run time.

What would be the benefit of porting RuneMaster from JS to Dyalog versus finishing up the current JS implementation?

**Reply by MaxCan-Code** [reply]
This project would fall under the category of [**Fun/Peripheral**](https://google.github.io/gsocguides/mentor/defining-a-project-ideas-list), meaning it isn't related to the current core development focus, but create new innovations and new perspective. Innovation being getting Dyalog to interact with other tools. Getting the Linux-only app to run under WSL and Docker on other platforms also demonstrates such interop.

**Comment by sloorush** [comment]
This project will not be a part of Dyalog's GSoC 2025 Project List
